deployment/helm/eclipse-ditto/values.yaml

```diff
@@ -2,135 +2,654 @@
+  ## ref: https://www.eclipse.org/ditto/installation-operating.html#configuration
+  systemProps: []
+  ## extraEnv to add arbritrary environment variable to concierge container
+  extraEnv: []
+  ## resources for the concierge container
+  resources: {}
+    # requests:
+    #   memory: 512Mi
+    # limits:
+    #   ## no cpu limit to avoid CFS scheduler limits
+    #   ## ref: https://doc.akka.io/docs/akka/snapshot/additional/deploy.html#in-kubernetes
+    #   ## cpu: ""
+    #   memory: 1Gi
+  ## readinessProbe configuration for concierge
+  ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes
+  readinessProbe:
+    initialDelaySeconds: 45
+    periodSeconds: 10
+    timeoutSeconds: 1
+    failureThreshold: 3
+  ## livenessProbe configuration for concierge
+  ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes
+  livenessProbe:
+    initialDelaySeconds: 180
+    periodSeconds: 10
+    timeoutSeconds: 3
+    failureThreshold: 4
+  networkPolicy:
+    ## enabled controls whether concierge related NetworkPolicy should be created
+    enabled: true
+  podDisruptionBudget:
+    ## enabled controls whether concierge related PodDisruptionBudget should be created
+    enabled: true
+    ## minAvailable number of replicas during voluntary disruptions
+    minAvailable: 1
+  ## nodeSelector
+  ## https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector
+  nodeSelector: {}
+  ## tolerations
+  ## ref: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
+  tolerations: []
+  ## affinity
+  ## ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
+  affinity: {}
+
+
+## ----------------------------------------------------------------------------
+## connectivity configuration
+## ref: https://www.eclipse.org/ditto/architecture-services-connectivity.html
 connectivity:
+  ## enabled controls whether connectivity related resources should be created
   enabled: true
-  name: connectivity
+  ## replicaCount for connectivity service
   replicaCount: 1
-  memory: 512Mi
-  networkPolicy:
-    enabled: true
+  ## updateStrategy for connectivity service
+  ## ref: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy
+  updateStrategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 0
+  ## additionalLabels on connectivity pods
+  additionalLabels: {}
+  ## additionalAnnotations on connectivity pods
+  additionalAnnotations: {}
   image:
+    ## repository for the connectivity docker image
     repository: docker.io/eclipse/ditto-connectivity
-    imagePullPolicy: IfNotPresent
-    tag: 1.0.0-M1a
-things:
-  enabled: true
-  name: things
-  replicaCount: 1
-  memory: 512Mi
+    ## tag for the connectivity docker image
+    tag: 1.0.0-M3
```

Comment:
Can we somehow define the Ditto version/tag to be used at only one place?
Can the `appVersion` be referenced here?

Reply:
No, there is no possibility to use templating / variables inside the values file.

Reply:
Unfortunately not (as @monotek correctly mentioned)
This is a tradeoff. You're right, version number is duplicated but this gives the flexibility of replacing single containers (e.g. for troubleshooting or bugfixing one might to deploy a customized container).